url56b1dc76d3_wufunxiang_Libs_tgz-pJ8-org_msgpack_util_TemplatePrecompilerJ8/src/org/msgpack/template/builder/beans/Statement.java

```diff
@@ -0,0 +1,610 @@
+                     * given Class object at first process only if the class
+                     * differs from Class itself
+                     */
+                    if (target != Class.class) {
+                        method = findMethod((Class<?>) target, methodName,
+                                arguments, true);
+                        result = method.invoke(null, arguments);
+                    }
+                } catch (NoSuchMethodException e) {
+                    // expected
+                }
+                if (method == null) {
+                    // static method was not found
+                    // try to invoke method of Class object
+                    if (BeansUtils.FORNAME.equals(methodName)
+                            && arguments.length == 1
+                            && arguments[0] instanceof String) {
+                        // special handling of Class.forName(String)
+                        try {
+                            result = Class.forName((String) arguments[0]);
+                        } catch (ClassNotFoundException e2) {
+                            result = Class.forName((String) arguments[0], true,
+                                    Thread.currentThread()
+                                            .getContextClassLoader());
+                        }
+                    } else {
+                        method = findMethod(targetClass, methodName, arguments,
+                                false);
+                        result = method.invoke(target, arguments);
+                    }
+                }
+            } else if (target instanceof Iterator<?>) {
+                final Iterator<?> iterator = (Iterator<?>) target;
+                final Method method = findMethod(targetClass, methodName,
+                        arguments, false);
+                if (iterator.hasNext()) {
+                    result = new PrivilegedAction<Object>() {
+                        public Object run() {
+                            try {
+                                method.setAccessible(true);
+                                return (method.invoke(iterator, new Object[0]));
+                            } catch (Exception e) {
+                                // ignore
+                            }
+                            return null;
+                        }
+
+                    }.run();
+                }
+            } else {
+                Method method = findMethod(targetClass, methodName, arguments,
+                        false);
+                method.setAccessible(true);
+                result = method.invoke(target, arguments);
+            }
+        } catch (InvocationTargetException ite) {
+            Throwable t = ite.getCause();
+            throw (t != null) && (t instanceof Exception) ? (Exception) t : ite;
+        }
+        return result;
+    }
+
+    private void arrayCopy(Class<?> type, Object[] src, Object dest, int length) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 17. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.